tests/queries/0_stateless/03533_lexer_c_library.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+clang++-19 -c -Os -fno-exceptions -fno-rtti -nostdlib -D LEXER_STANDALONE_BUILD "${CUR_DIR}/../../../src/Parsers/Lexer.cpp" -o "${CLICKHOUSE_TMP}/Lexer.o"
+
+> main.c echo '
+extern unsigned long clickhouse_lexer_size;
+void clickhouse_lexer_create(void * ptr, const char * begin, const char * end, unsigned long max_query_size);
+unsigned char clickhouse_lexer_next_token(void * ptr, const char ** out_token_begin, const char ** out_token_end);
+int clickhouse_lexer_token_is_significant(unsigned char token);
+int clickhouse_lexer_token_is_error(unsigned char token);
+int clickhouse_lexer_token_is_end(unsigned char token);
```

Comment:
it's maybe confusing to use it this way - probably we need some header with this API